doc/man3/OSSL_PARAM_TYPE.pod

```diff
@@ -294,12 +411,23 @@ could fill in the parameters like this:
     const OSSL_PARAM *p;
 
     if ((p = OSSL_PARAM_locate(params, "foo")) == NULL)
-        OSSL_PARAM_set_utf8_ptr(p, "foo value");
+        error();
+    OSSL_PARAM_set_utf8_ptr(p, "foo value");
     if ((p = OSSL_PARAM_locate(params, "bar")) == NULL)
-        OSSL_PARAM_set_utf8_ptr(p, "bar value");
-    if ((p = OSSL_PARAM_locate(params, "cookie")) == NULL)
+        error();
+    OSSL_PARAM_set_utf8_ptr(p, "bar value");
+    if ((p = OSSL_PARAM_locate(params, "cookie")) != NULL)
         OSSL_PARAM_set_utf8_ptr(p, "cookie value");
```

Comment:
I disagree with the error, but good catch on the NULL check.  The code should be:

``` C
    if ((p = OSSL_PARAM_locate(params, "foo")) != NULL)
        OSSL_PARAM_set_utf8_ptr(p, "foo value");
    if ((p = OSSL_PARAM_locate(params, "bar")) != NULL)
        OSSL_PARAM_set_utf8_ptr(p, "bar value");
    if ((p = OSSL_PARAM_locate(params, "cookie")) != NULL)
        OSSL_PARAM_set_utf8_ptr(p, "cookie value");
```

The rationale is still the same, if the other end is only interested in the cookie value, *why should it be obligated to also get the foo and bar values*?

Reply:
Perhaps, the RSA example would work better. _n_ and _e_ being mandatory, the rest being optional?

Reply:
If the caller is only interested in getting the values of *p* and *q* (if those are accessible)?